File: 392.Is Subsequence/is_subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int lpt = 0; 
        int tpt = 0;

        while(lpt < s.size() && tpt < t.size()) {
            if(s[lpt] == t[tpt]) lpt++; 

            tpt++;
        }
        if(lpt == s.size()) return true; 
        return false;
    }
};

File: 392.Is Subsequence/is_subsequence.py
class Solution(object):
    def isSubsequence(self, s, t):
        lpt, tpt = 0, 0
            
        while tpt < len(t) and lpt < len(s):
            if t[tpt] == s[lpt]:
                lpt+=1 

            tpt += 1
        return lpt == len(s)